Cirrious/Cirrious.MvvmCross.Droid.RecyclerView/MvxRecyclerView.cs

```diff
@@ -0,0 +1,139 @@
+            : base(context, attrs, defStyle)
+        {
+            _gestureDetector = new GestureDetector(context, new GestureListener());
+            this.AddOnItemTouchListener(this);
+
+            // Note: Any calling derived class passing a null adapter is responsible for setting
+            // it's own itemTemplateId
+            if (adapter == null)
+                return;
+
+            SetLayoutManager(new LinearLayoutManager(context));
+
+            var itemTemplateId = MvxAttributeHelpers.ReadListItemTemplateId(context, attrs);
+            adapter.ItemTemplateId = itemTemplateId;
+            Adapter = adapter;
+        }
+
+        #endregion
+
+        public sealed override void AddOnItemTouchListener(IOnItemTouchListener listener)
+        {
+            base.AddOnItemTouchListener(listener);
+        }
+        public sealed override void SetLayoutManager(LayoutManager layout)
+        {
+            base.SetLayoutManager(layout);
+        }
+
+        public new IMvxRecyclerAdapter Adapter
+        {
+            get { return base.GetAdapter() as IMvxRecyclerAdapter; }
+            set
+            {
+                var existing = Adapter;
+
+                if (existing == value)
+                    return;
+
+                // Support lib doesn't seem to have anything similar to IListAdapter yet
+                // hence cast to Adapter.
+
+                if (value != null && existing != null)
+                {
+                    value.ItemsSource = existing.ItemsSource;
+                    value.ItemTemplateId = existing.ItemTemplateId;
+
+                    SwapAdapter((Adapter)value, false);
+                }
+                else
+                {
+                    SetAdapter((Adapter)value);
+                }
+            }
+        }
+
+        [MvxSetToNullAfterBinding]
+        public IEnumerable ItemsSource
+        {
+            get { return Adapter.ItemsSource; }
+            set { Adapter.ItemsSource = value; }
+        }
+
+        public int ItemTemplateId
+        {
+            get { return Adapter.ItemTemplateId; }
+            set { Adapter.ItemTemplateId = value; }
+        }
+
+        public ICommand ItemClick { get; set; }
+
+        protected virtual void ExecuteCommandOnItem(ICommand command, int position)
+        {
+            if (command == null)
+                return;
+
+            var item = Adapter.GetItem(position);
+
+            if (item == null || !command.CanExecute(item))
+                return;
+
+            command.Execute(item);
+        }
+
+        #region IOnItemTouchListener
+
+        public virtual bool OnInterceptTouchEvent(Android.Support.V7.Widget.RecyclerView view, MotionEvent e)
+        {
+            View childView = view.FindChildViewUnder(e.GetX(), e.GetY());
+            if (childView != null && _gestureDetector.OnTouchEvent(e))
+            {
+                PlaySoundEffect(SoundEffects.Click);
```

Comment:
You might be right! ListView does this - as do many of the other views however with this implementation of intercepting the touch event I found the view doesn't do this hence the sound effect!

For reference, this is `View.performClick()` calling `playSoundEffect(SoundEffectConstants.CLICK);` in the Android source.
https://github.com/android/platform_frameworks_base/blob/master/core/java/android/view/View.java#L4779